Lib/fontTools/varLib/__init__.py

```diff
@@ -313,13 +317,17 @@ def _add_gvar(font, model, master_ttfs, tolerance=0.5, optimize=True):
 		endPts = control[1] if control[0] >= 1 else list(range(len(control[1])))
 
 		for i,(delta,support) in enumerate(zip(deltas[1:], supports[1:])):
-			if all(abs(v) <= tolerance for v in delta.array):
+			if all(abs(v) <= tolerance for v in delta.array) and not isComposite:
 				continue
 			var = TupleVariation(support, delta)
 			if optimize:
 				delta_opt = iup_delta_optimize(delta, origCoords, endPts, tolerance=tolerance)
 
 				if None in delta_opt:
+					# In composite glyphs, there should be one 0 entry
+					# to make sure the gvar entry is written to the font.
```

Comment:
it's not clear from this comment why "in composite glyphs, there should be one 0 entry to make sure the gvar entry is written to the font". Have we established whether this is a requirement in the OT spec or implementation detail (or even a bug) in CoreText?

Reply:
The spec is not very explicit about if glyphs (composites) that are not referenced in `gvar` should have variations applied (through component glyph variations).

The closest statement I could find is ‘Note: inferring of deltas for un-referenced points applies only to simple glyphs, not to composite glyphs.’

This could be interpreted in a way that composite glyphs must have explicit `gvar` entries if variations should be applied to them.

Reply:
btw, what does Windows do?

Reply:
There is no problem on Window (10, tested in Edge) if the `gvar` entries are not present.

Reply:
> The closest statement I could find is [‘Note: inferring of deltas for un-referenced points applies only to simple glyphs, not to composite glyphs.’](https://docs.microsoft.com/de-de/typography/opentype/spec/gvar#inferred-deltas-for-un-referenced-point-numbers)
> 
> This could be interpreted in a way that composite glyphs must have explicit `gvar` entries if variations should be applied to them.

That cannot be deduced from that, no.   What that means simply is that if a point has no deltas in gvar for composite glyphs, then the value should be assumed to be zero.

Don't commit this unless you can produce a test and this fixes.

Reply:
OK ... so do we agree the different rendering is a macOS bug and we don’t want a (partial) workaround?